dev_guide/pod_preset.adoc

```diff
@@ -0,0 +1,565 @@
+[[dev-guide-pod-presets]]
+= Injecting Information into Pods Using Pod Presets
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+
+A _pod preset_ is an object that injects user-specified information into pods as they are created.
+
+//tag::pod-preset[]
+[IMPORTANT]
+====
+Pod presets is a Technology Preview feature only.
+ifdef::openshift-enterprise[]
+Technology Preview features are not
+supported with Red Hat production service level agreements (SLAs), might not be
+functionally complete, and Red Hat does not recommend to use them for
+production. These features provide early access to upcoming product features,
+enabling customers to test functionality and provide feedback during the
+development process.
+
+For more information on Red Hat Technology Preview features support scope, see
+https://access.redhat.com/support/offerings/techpreview/.
+endif::[]
+====
+
+Using pod preset objects you can inject:
+
+* xref:../dev_guide/secrets.adoc#dev-guide-secrets[secret objects]
+* xref:../dev_guide/configmaps.adoc#consuming-configmap-in-pods[`ConfigMap` objects]
+* xref:../dev_guide/volumes.adoc#dev-guide-volumes[storage volumes]
+* container volume mounts
+* environment variables
+
+Developers only need to add the appropriate label to the pod specification in order to add all that information to the pod. The xref:../architecture/core_concepts/pods_and_services.adoc#labels[label] on a pod associates the pod with one or more pod preset objects that have a matching xref:../architecture/core_concepts/pods_and_services.adoc#services[label selectors].
+
+Using pod presets, a developer can provision pods without needing to know the details about the service the pod will consume. An administrator can keep configuration items of a service invisible from a developer without preventing the developer from deploying pods. For example, an administrator can create a pod preset that provides the name, user name, and password for a database through a secret object and the database port through an environment variable. The pod developer only needs to know the label to use to include all the information in pods. A developer can also create pod presets and perform all the same tasks. For example, the developer can create a preset that injects environment variable automatically into multiple pods.
+
+When a pod preset is applied to a pod, {product-title} modifies the pod specification, adding the injectable data and annotating the pod spec to show that it was modified by a pod preset. The annotation is of the form:
+
+----
+podpreset.admission.kubernetes.io/<pod-preset name>: `resource version`
+----
+
+You can configure a pod specification to prevent this modification using the `podpreset.admission.kubernetes.io/exclude` annotation in the pod specification, as shown in the pod specification example below.
```

Comment:
Might be worth mentioning the key needs to be set to "true", since I don't think it'll take effect without it.